src/MultiAgentReinforcementLearning/src/policies/multi_agent.jl

```diff
@@ -0,0 +1,56 @@
+export MultiAgentManager
+
+Base.@kwdef mutable struct MultiAgentManager{T <: AbstractPolicy} <: AbstractPolicy
+    agents::Dict{S,T} where {S<: Union{String, Symbol, Integer}}
+    cur_player::Union{String, Symbol, Integer} = first(keys(agents))
+end
+
+Base.getindex(A::MultiAgentManager, x) = getindex(A.agents, x)
+
+"""
+    MultiAgentManager(player => policy...)
+
+This is the simplest form of multiagent system. At each step they observe the
+environment from their own perspective and get updated independently.
+"""
+MultiAgentManager(policies...) =
+    MultiAgentManager(Dict{Any,Any}(nameof(p) => p for p in policies))
+
+RLBase.prob(A::MultiAgentManager, env::AbstractEnv, args...) = prob(A[A.cur_player].policy, env, args...)
+
+(A::MultiAgentManager)(env::AbstractEnv) = A(env, DynamicStyle(env))
+
+function (A::MultiAgentManager)(env::AbstractEnv, ::Sequential)
+    while current_player(env) == chance_player(env)
+        env |> legal_action_space |> rand |> env
+    end
+    return A[A.cur_player](env)
```

Comment:
It seems weird that in the `while` loop we fetch `current_player(env)`, but in the return statement we use `A.cur_player`, is there any reason for this inconsistency?

Reply:
```suggestion
    A.cur_player = current_player(env)
    return A[A.cur_player](env)
```

Reply:
I'm now really confused...what is `current_player(env)` tracking vs `A.cur_player` from the manager? Seems like unnecessary duplication? Or a naming issue?

Reply:
In order to clarify it: `current_player(env)` is the agent which is going to take the next action. That being said, it means that after passing the action to the environment, it changes. However, `MultiAgentManager ` should update the elements of an agent's trajectory in the same iteration it took the action and passed it to the environment. Otherwise, the collected reward is a time shifted version and for the independent method this would mean that false reward signals would be present. 
The agent would not be able to learn an optimal policy (if that is possible for independent methods in general). 
Does that clear your confusion, or do I miss a point here?

Reply:
Thanks for your clarification, it really helped! What I wasn't really paying attention to is the degree to which the new `multiagent` setup differs from the old version, e.g. [here](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/v0.10.0/src/ReinforcementLearningCore/src/policies/agents/multi_agent.jl). I (personally) don't love that we now carry around the current player in two different places (e.g. `env` and `MultiAgent`) and that `chance_player` is being used wherever the `Sequential` trait exists...AFAIK, `chance_player` should only need to be implemented for non-`Deterministic` environments, but I don't see where this is supported in the new code.